### PR TITLE
Update for PHP 7.2 or later

### PR DESCRIPTION
**Pull request by kenjis**
Supersedes #12

Needs:
- [x] updated Aura.Project_Kernel https://github.com/auraphp/Aura.Project_Kernel/pull/15
- [x] updated Aura.Cli https://github.com/auraphp/Aura.Cli/pull/65
- [ ] Aura.Project_Kernel 4.0.0 release
- [ ] Aura.Cli 4.0.0 release

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,11 +15,18 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
-        "aura/project-kernel": "~2.0",
-        "aura/cli": "~2.0",
+        "php": ">=7.2.0",
+        "aura/project-kernel": "4.x-dev",
```
Comment (harikt) [inline composer.json:19]: I think you should use dev-4.x . The github actions is failing due to composer issue.

**Comment by harikt** [comment]
Regarding the coverage report and scrutinizer, I want to dig into it.  I tried sometime experimenting with it, but seems some issues. I will checkout that later.

File: composer.json (hunk 1 above)
Comment (kenjis) [reply composer.json:19]:
```diff
$ git diff
diff --git a/composer.json b/composer.json
index ecc1169..8635e0f 100644
--- a/composer.json
+++ b/composer.json
@@ -16,7 +16,7 @@
     ],
     "require": {
         "php": ">=7.2.0",
-        "aura/project-kernel": "4.x-dev",
+        "aura/project-kernel": "dev-4.x",
         "aura/cli": "update-for-php7-dev",
         "aura/dispatcher": "~2.0"
     },
```

```sh-session
$ composer update
Loading composer repositories with package information                                                         Updating dependencies                                 
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires aura/project-kernel dev-4.x, found aura/project-kernel[dev-coverage, dev-master, dev-develop-2, 2.0.0-beta1, ..., 2.x-dev, 4.x-dev] but it does not match the constraint.
...
```

Comment (kenjis) [reply composer.json:19]: @harikt `4.x-dev` is correct. The github action fail is because it is a branch.
We should use `^4.0`. We need 4.0.0 release.

**Comment by kenjis** [comment]
@pmjones @harikt @koriym 
Could you release Aura.Project_Kernel 4.0.0 and Aura.Cli 4.0.0?
This PR needs them as dependencies.

**Comment by harikt** [comment]
I will check the test failure.

File: composer.json (hunk 1 above)
Comment (harikt) [reply composer.json:19]: Great. I got a bit confused.

**Comment by harikt** [comment]
The reasons for failures were 

1. composer validate --strict  . I have commented out this for now.
2. composer install --prefer-dist --no-progress.  . Removed prefer distribution as we currently don't have a release.

You asked for a release only because of the installation failure. Am I right?

**Comment by kenjis** [comment]
@harikt 
> You asked for a release only because of the installation failure. Am I right?

Yes, at the moment.

But I want to update the versions to `^4.0` in the composer.json:
https://github.com/auraphp/Aura.Cli_Kernel/blob/362fba59d96d650985bc70475da191909b4db385/composer.json#L19-L20

**Comment by harikt** [comment]
I will look into this in the weekend.

**Comment by kenjis** [comment]
@harikt Thank you!

`4.x-dev` is not *stable*. So it is a pain when I use this package.
I need to change the stability setting or list all dependent unstable packages in the `require` section.

**Comment by harikt** [comment]
Hey @kenjis ,

I made 4.x alpha releases for both Aura.Cli and Aura.Project_Kernel.

**Pull request merged**